browser/brave_cosmetic_resources_tab_helper.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_BRAVE_COSMETIC_RESOURCES_TAB_HELPER_H_
+#define BRAVE_BROWSER_BRAVE_COSMETIC_RESOURCES_TAB_HELPER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/memory/weak_ptr.h"
+#include "brave/content/browser/cosmetic_filters_observer.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+class BraveCosmeticResourcesTabHelper
+    : public content::CosmeticFiltersObserver,
+      public content::WebContentsObserver,
+      public content::WebContentsUserData<BraveCosmeticResourcesTabHelper>,
+      public base::SupportsWeakPtr<BraveCosmeticResourcesTabHelper> {
```

Comment:
btw I'd suggest to switch from `SupportsWeakPtr` to using `base::WeakPtrFactory` as it does more checks and is more common pattern

Reply:
done